engine/src/test/java/com/redhat/cloud/notifications/processors/email/EmailAggregatorTest.java

```diff
@@ -98,31 +114,49 @@ void shouldTestRecipientsFromSubscription() {
         assertNotNull(result);
         assertTrue(result.size() == 1);
         assertTrue(result.keySet().stream().filter(usr -> usr.getEmail().equals("user-1")).count() == 1);
+        User user = result.keySet().stream().findFirst().get();
+        assertEquals(4, ((LinkedHashMap) result.get(user).get("policies")).size());
+        verify(emailAggregationRepository, times(1)).getEmailAggregation(any(EmailAggregationKey.class), any(LocalDateTime.class), any(LocalDateTime.class), anyInt(), anyInt());
+        verify(emailAggregationRepository, times(1)).getEmailAggregation(any(EmailAggregationKey.class), any(LocalDateTime.class), any(LocalDateTime.class), eq(0), eq(emailAggregator.aggregationMaxPageSize));
+        statelessSessionFactory.withSession(statelessSession -> {
+            emailAggregationRepository.purgeOldAggregation(aggregationKey, LocalDateTime.now());
+        });
+        reset(emailAggregationRepository); // just reset mockito counter
 
         // Test user subscription based on event type
         featureFlipper.setUseEventTypeForSubscriptionEnabled(true);
         result = aggregate();
+        verify(emailAggregationRepository, times(1)).getEmailAggregation(any(EmailAggregationKey.class), any(LocalDateTime.class), any(LocalDateTime.class), anyInt(), anyInt());
+        verify(emailAggregationRepository, times(1)).getEmailAggregation(any(EmailAggregationKey.class), any(LocalDateTime.class), any(LocalDateTime.class), eq(0), eq(emailAggregator.aggregationMaxPageSize));
+        reset(emailAggregationRepository); // just reset mockito counter
 
         // nobody subscribed to the right event type yet
-        assertTrue(result.size() == 0);
+        assertEquals(0, result.size());
 
         resourceHelpers.createEventTypeEmailSubscription("org-1", "user-2", eventType1, DAILY);
+        // because on previous step, nobody subscribed to the rignt event type, we already have 4 records on database
```

Comment:
Is that really the reason? From what I can see in the code, `EmailAggregator#getAggregated` does not purge the DB data so even if someone subscribed to the event type, we would still have the 4 records in the DB here.

Reply:
This would sound more accurate to me:
```suggestion
        // because after the previous aggregate() call the email_aggregation DB table was not purged, we already have 4 records on database
```

Reply:
Right